precompiles/stateful/Vesting.sol

```diff
@@ -0,0 +1,138 @@
+// SPDX-License-Identifier: LGPL-3.0-only
+pragma solidity >=0.8.18;
+
+import "../common/Types.sol";
+
+/// @dev The VestingI contract's address.
+address constant VESTING_PRECOMPILE_ADDRESS = 0x0000000000000000000000000000000000000803;
+
+/// @dev The VestingI contract's instance.
+VestingI constant VESTING_CONTRACT = VestingI(VESTING_PRECOMPILE_ADDRESS);
+
+/// @dev Define all the available staking methods.
+string constant MSG_CREATE_CLAWBACK_VESTING_ACCOUNT = "/evmos.vesting.v2.MsgCreateClawbackVestingAccount";
+string constant MSG_FUND_VESTING_ACCOUNT = "/evmos.vesting.v2.MsgFundVestingAccount";
+string constant MSG_CLAWBACK = "/evmos.vesting.v2.MsgClawback";
+string constant MSG_CONVERT_VESTING_ACCOUNT = "/evmos.vesting.v2.MsgConvertVestingAccount";
+string constant MSG_UPDATE_VESTING_FUNDER = "/evmos.vesting.v2.MsgUpdateVestingFunder";
+
+
+// Period defines a length of time and amount of coins that will vest.
+struct Period {
+    int64 length;
+    Coin[] amount;
+}
+
+/// @author Evmos Team
+/// @title Vesting Precompiled Contract
+/// @dev The interface through which solidity contracts will interact with vesting.
+/// We follow this same interface including four-byte function selectors, in the precompile that
+/// wraps the pallet.
```

Comment:
What are we referring to when we say pallet? Is it something that references Moonbeam precompiles?